### PR TITLE
[Doc] Add valid micro-service details

### PR DESCRIPTION
**Pull request by xiguiw**
## Description

Add valid micro-service details. 
The GenAI build mega-service on top of micro service and server.

This DOC describe the steps to check and validate the mciro-servcie and server in detail.

### CODE REVIEW

**Comment by Spycsh** [comment]
@xiguiw , thanks for the PR. Please modify the Docker image names to be consistent with https://github.com/opea-project/GenAIExamples/commit/8ae28ebfdec1472a0d030694023f226d7f9a056d.

File: valid_service.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# How to Check and Valid Micro Service in the GenAI Example
+
+GenAI examples build mega-services on top of the micro-service and server.
+To make mega-service works, each micro service and server must work as expected.
+
+Here are steps to check the micro service and server.
+
+## Check environment variables
+
+1. Make sure environment variables are set
```
Comment (Spycsh) [inline valid_service.md:10]:
```
## 1. Check environment variables

Make sure environment variables are set
```

should be better.

File: valid_service.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+echo $REDIS_URL
+echo $INDEX_NAME
+echo $HUGGINGFACEHUB_API_TOKEN
+echo $MEGA_SERVICE_HOST_IP
+echo $BACKEND_SERVICE_ENDPOINT
+```
+
+## Check docker container
+
+2. Check the docker container are started
```
Comment (Spycsh) [inline valid_service.md:31]: Same here.

File: valid_service.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+```
+115f81d80017   ghcr.io/huggingface/tgi-gaudi:1.2.1                     "text-generation-lau…"   5 hours ago   Exited (2) 5 hours ago                                                                                          tgi-gaudi-server
+7eb49377b256   ghcr.io/huggingface/text-embeddings-inference:cpu-1.2   "text-embeddings-rou…"   5 hours ago   Exited (2) 5 hours ago
+```
+
+To get the container status, next check the container logs.
+
+## Check docker container log
+
+3. Check the log of container
```
Comment (Spycsh) [inline valid_service.md:62]: Same here.

File: valid_service.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+(base) ubuntu@gaudi-vm:~$ docker logs -t 7eb49377b256
+2024-05-24T02:25:59.815764340Z error: a value is required for '--model-id <MODEL_ID>' but none was supplied
+2024-05-24T02:25:59.816762396Z
+2024-05-24T02:25:59.816769350Z For more information, try '--help'.
+```
+
+It indicates the MODLE_ID is not set correctly. Check model_id environment variables are set correctly, make sure environment are spelled correctly.
+Then restart the containers.
+
+## Check Each service of the mega service
```
Comment (Spycsh) [inline valid_service.md:82]: `## 4. Check Each service of the mega service`

File: valid_service.md (hunk 3 above)
Comment (Spycsh) [reply valid_service.md:62]: Also add another command to check the overall log:

`docker compose -f docker_compose.yaml logs`

File: valid_service.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+2. Check the docker container are started
+
+`docker ps -a`
+
+For example, the ChatQnA example starts 9 servers, check these 9 servers all running. i.e, the contanier `STATUS` are `Up`
+
+```
+(base) ubuntu@gaudi-vm:~$ docker ps -a
+CONTAINER ID   IMAGE                                                   COMMAND                  CREATED       STATUS                   PORTS                                                                                  NAMES
+92721ba5430a   opea/gen-ai-comps:chatqna-ui-server                     "docker-entrypoint.s…"   5 hours ago   Up 10 minutes            0.0.0.0:5173->5173/tcp, :::5173->5173/tcp                                              chatqna-gaudi-ui-server
```
Comment (Spycsh) [inline valid_service.md:40]: `opea/chatqna-ui:latest` name refactor, please check https://github.com/opea-project/GenAIExamples/commit/8ae28ebfdec1472a0d030694023f226d7f9a056d

**Review by Spycsh — CHANGES_REQUESTED** [review]

**Review by Spycsh — APPROVED** [review]

**Comment by lvliang-intel** [comment]
@xiguiw  Please rebase your code and fix the conflict. And also test with latest code and then update the document.

**Comment by xiguiw** [comment]
> @xiguiw Please rebase your code and fix the conflict. And also test with latest code and then update the document.

Updated. 
Test with latest code.

@lvliang-intel  @Spycsh  Please help to review. 

Thanks!

**Comment by chensuyue** [comment]
Please fix the spell issue,  
how_to_valid_service.md:60: happend ==> happened, happens, happen

**Comment by xiguiw** [comment]
> Please fix the spell issue, how_to_valid_service.md:60: happend ==> happened, happens, happen

Done. Thanks!

**Review by Spycsh — APPROVED** [review]

File: how_to_valid_service.md [hunk 6]
```diff
@@ -0,0 +1,405 @@
+# How to Check and Valid Micro Service in the GenAI Example
```
Comment (chensuyue) [inline how_to_valid_service.md:1]: Valid -> Validate?  
Suggest to put this doc under `ChatQnA/docker/gaudi`.

Comment (xiguiw) [reply how_to_valid_service.md:1]: Validate micro service are common to all examples (and there is no difference between Xeon and Gaudi).
So put it on a common directory for all examples. 

@chensuyue Is there other directory recommened?

**Review by chensuyue — APPROVED** [review]

**Pull request merged**